repository_cli/cli/records.py

```diff
@@ -382,66 +382,94 @@ def replace_identifier(identifier: str, pid: str):
     click.secho(f"Identifier for '{pid}' replaced.", fg="green")
 
 
-@rdmrecords.command('add_file')
-@click.argument('recid', type=str)
-@click.argument('fp', type=click.File('rb'))
-@click.option('--replace-existing', '-f', is_flag=True, default=False)
+@rdmrecords.command("add_file")
+@click.argument("recid", type=str)
+@click.argument("fp", type=click.File("rb"))
+@click.option("--replace-existing", "-f", is_flag=True, default=False)
+@click.option("--data-model", default="rdm")
 @with_appcontext
-def add_file(recid, fp, replace_existing):
+def add_file(recid, fp, replace_existing, data_model):
     """Add a new file to a published record."""
-    from invenio_db import db
-    from invenio_access.permissions import system_identity
-    record = get_records_service().read(identity=system_identity, id_=recid)._record
+    identity = get_identity("system_process", role_name="admin")
+    service = get_records_service(data_model=data_model)
 
+    try:
+        record = service.read(identity=identity, id_=recid)._record
+    except PIDDoesNotExistError as e:
+        click.echo(
+            click.style(
```

Comment:
Minor: the function [click.secho](https://click.palletsprojects.com/en/8.1.x/api/#click.secho) exists which would combine click.echo with click.style. This would make the code less verbose